enhancements/sdk-generate-commands.md

```diff
@@ -0,0 +1,411 @@
+[bundle][bundle-format] and [package manifests][package-manifests-format] format
+generation scenarios with a plethora of flags. The number of flags and confusing
+command name (`generate csv` manages higher-level formats) creates non-ideal UX.
+This proposal details new workflows for bundle and package manifests formats
+for old and [new][ep-kb-integration] `operator-sdk`-build Operator projects.
+
+To clearly define what old and new layouts mean:
+* Old project layouts: scaffolded with `operator-sdk new`
+* New project layouts: scaffolded with `operator-sdk init`
+
+We plan to improve the developer experience for operator authors using Operator SDK by providing:
+1. One `operator-sdk` subcommand _per format_ for generating manifests and metadata.
+1. Interactive-by-default commands to get UI information for packaged manifests and metadata.
+
+[bundle-format]:https://github.com/operator-framework/operator-registry/blob/master/docs/design/operator-bundle.md#bundle-manifest-format
+[package-manifests-format]:https://github.com/operator-framework/operator-registry/tree/v1.5.3/#manifest-format
+[ep-kb-integration]:https://github.com/operator-framework/operator-sdk/blob/master/proposals/kubebuilder-integration.md
+
+## Motivation
+
+Currently, the `operator-sdk generate csv` command is used for generating CSVs,
+which scaffolds either a bundle or package manifests format and writes a
+[`ClusterServiceVersion` (CSV)][doc-csv] manifest to disk depending on the format selected.
+This process requires a developer to manually edit the CSV when any required
+field is missing, or if they want to add certain pieces of [UI metadata][doc-csv-metadata].
+Furthermore selecting the format is non-obvious; adding to the confusion is
+the fact that the operator-framework has no plans to deprecate the package manifests format
+in the near future, so having documentation for both formats in one command is confusing.
+
+[doc-csv]:https://github.com/operator-framework/operator-lifecycle-manager/blob/0.14.2/doc/design/building-your-csv.md
+[doc-csv-metadata]:https://github.com/operator-framework/operator-lifecycle-manager/blob/0.14.2/doc/design/building-your-csv.md#your-custom-resource-definitions
+
+## Goals
+
+* Simplify UX for generating bundle and package manifests formats.
+* Delineate bundle and package manifests format generation and usage.
+* _(New project layouts)_ Use a Makefile wherever possible.
+
+## Non-Goals
+
+* Deprecate or remove package manifests support from the Operator SDK.
+* Discuss the `run` subcommand in relation to bundles or package manifests.
+
+## Proposal
+
+There are two parts to this proposal:
+
+1. Add `generate bundle` and `generate packagemanifests` subcommands for both
+old and new project layouts.
+1. _(New project layouts)_ Add Makefile recipes `make bundle` and `make bundle-build`
+that invoke `generate bundle` and `docker build -f bundle.Dockerfile .`, respectively.
+
+### User Stories
+
```

Comment:
I am as an Operator Developer would like to use the SDK tool to integrate my Kubebuiler project with OLM

Reply:
Out of scope. This is only discussing UX of `operator-sdk` subcommands.

Reply:
We can not address the above story in the PR
However, IMO we cannot do an implementation that will NOT allow we to address it.

Reply:
IMO this doc describes needs that go beyond the UX and how the bundle commands should be implemented for the projects with the new layout.  

And then, I understand that it is an important story to achieve the SDK business strategy which motivates the KB integration. So, if we are here describing and defining how the implementation of this command should be we should take into account this requirement.

Reply:
That story is too general. I could see a story like
```
As an Operator developer of a vanilla kubebuilder project, I would like documentation on how to add bundle/package generation to my project
```

Reply:
IMO:
- We have motivations to integrate the SDK with Kubebuilder that we need to keep in mind to ensure that we will be able to achieve them and we will not add complexities to do that. 
- This doc is specific to the **new OLM commands with KB integration** so, the story is very specific to this context and not any other tool and in this way, I do not consider general at all. 

Note that, by not adding this story we are doing definitions here that would make harder/complexity we achieve this end goal. I am afraid that it will lead us a solution proposed as doc how to change the Makefile and projects layout to be able to do it when it could not necessarily be required at all.